### PR TITLE
migrate c4core to conan v2

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
This PR was auto-generated by a bash script. Most of the recipes (around 80%) can be built. It this one is failing, feel free to contribute a manual fix.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/c4core/all/conanfile.py [hunk 1]
```diff
@@ -80,9 +81,9 @@ def package(self):
         self.copy(pattern="LICENSE*", dst="licenses", src=self._source_subfolder)
         cmake = self._configure_cmake()
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "cmake"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "include"), "*.natvis")
+        tools.files.rmdir(self, os.path.join(self.package_folder, "cmake"))
+        tools.files.rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        tools.files.rm(self, os.path.join(self.package_folder, "include"), "*.natvis")
```
Comment (uilianries) [inline recipes/c4core/all/conanfile.py:86]: @AndreyMlashkin It's inverted. The pattern should come first, before the path.

Comment (uilianries) [inline recipes/c4core/all/conanfile.py:86]: @AndreyMlashkin It's inverted. The pattern should come first, before the path.

File: recipes/c4core/all/test_package/conanfile.py [hunk 2]
```diff
@@ -12,6 +13,6 @@ def build(self):
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
+        if not tools.build.cross_building(self, self):
```
Comment (uilianries) [inline recipes/c4core/all/test_package/conanfile.py:16]:
```suggestion
        if not tools.build.cross_building(self):
```

File: recipes/c4core/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/c4core/all/conanfile.py:86]: Anyway, this script rocks!

Comment (AndreyMlashkin) [reply recipes/c4core/all/conanfile.py:86]: ok, thanks. I will fix my script and re-interate

**Pull request closed without merging**